src/thunderbots/software/test/primitive/primitive.cpp

```diff
@@ -51,17 +55,79 @@ TEST(PrimitiveTest, create_primitive_from_message_test)
 
     thunderbots_msgs::Primitive prim_message = move_prim.createMsg();
 
-    std::unique_ptr<Primitive> new_prim = MovePrimitive::createPrimitive(prim_message);
+    std::unique_ptr<Primitive> new_prim = Primitive::createPrimitive(prim_message);
 
-    std::vector<double> param_array = new_prim->getParameterArray();
+    // Since we have a `Primitive` and NOT a `MovePrimitive`, we have to check that the
+    // values are correct by getting them from the generic parameter array
+    std::vector<double> params = new_prim->getParameters();
 
     EXPECT_EQ("Move Primitive", new_prim->getPrimitiveName());
     EXPECT_EQ(robot_id, new_prim->getRobotId());
-    EXPECT_DOUBLE_EQ(destination.x(), param_array[0]);
-    EXPECT_DOUBLE_EQ(destination.y(), param_array[1]);
-    EXPECT_DOUBLE_EQ(final_angle.toRadians(), param_array[2]);
-    EXPECT_DOUBLE_EQ(final_speed, param_array[3]);
-    EXPECT_EQ(new_prim->getExtraBitArray(), std::vector<bool>());
+    EXPECT_DOUBLE_EQ(destination.x(), params[0]);
+    EXPECT_DOUBLE_EQ(destination.y(), params[1]);
+    EXPECT_DOUBLE_EQ(final_angle.toRadians(), params[2]);
+    EXPECT_DOUBLE_EQ(final_speed, params[3]);
+    EXPECT_EQ(new_prim->getExtraBits(), std::vector<bool>());
+}
+
+// Test that we can correctly translate a ChipPrimitive like:
+// `Primitive` -> `ROS Message` -> `Primitive` and get back the same primitive we
+// started with
+TEST(PrimitiveTest, create_ChipPrimitive_from_message_test)
```

Comment:
test name could be more descriptive, eg. `convert_ChipPrimitive_to_message_and_back_to_ChipPrimitive`. Basically something to indicate we are doing _two_ conversions, not just a single `message -> ChipPrimitive` as the name implies.

Reply:
Renamed all tests as per the above recommendation.